### PR TITLE
Permit the use of suspend on a method returning Flow

### LINKED ISSUE #20345: Resteasy Reactive: Kotlin `Flow` and `Multi` are not supported in suspend functions
### Describe the bug

`Flow` and `Multi` cannot be returned from a suspendable function (resteasy reactive).

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

1. https://code.quarkus.io/?e=resteasy-reactive-jackson&e=kotlin
2. Add the following code:

```kotlin
import io.smallrye.mutiny.Multi
import io.smallrye.mutiny.coroutines.asMulti
import kotlinx.coroutines.delay
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.flow
import javax.enterprise.context.ApplicationScoped
import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType

@ApplicationScoped
@Path("/flow")
class FlowResource {

    @GET
    @Path("/1")
    @Produces(MediaType.TEXT_PLAIN)
    suspend fun flow1Method(): Flow<String> { // kotlinx.coroutines.flow.SafeFlow@6b336f3a
        compute()

        return flow {
            emit("1")
            emit("2")
            emit("3")
        }
    }

    @GET
    @Path("/2")
    @Produces(MediaType.TEXT_PLAIN)
    suspend fun multiMethod(): Multi<String> { // io.smallrye.mutiny.operators.multi.builders.EmitterBasedMulti@5eec36c2
        compute()

        return flow {
            emit("1")
            emit("2")
            emit("3")
        }.asMulti()
    }

    private suspend fun compute() {
        delay(1000)
    }
}
```

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

Java 11

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.2.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven 3.8.1

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #20345

### CODE REVIEW

File: extensions/resteasy-reactive/quarkus-resteasy-reactive-kotlin/deployment/src/main/java/io/quarkus/resteasy/reactive/kotlin/deployment/KotlinCoroutineIntegrationProcessor.java [hunk 1]
```diff
@@ -185,4 +191,10 @@ public boolean isMethodSignatureAsync(MethodInfo info) {
             }
         });
     }
+
+    private static HandlerChainCustomizer flowCustomizer() {
+        return new FixedHandlersChainCustomizer(
+                List.of(new FlowToPublisherHandler(), new PublisherResponseHandler()),
```
Comment (evanchooly) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive-kotlin/deployment/src/main/java/io/quarkus/resteasy/reactive/kotlin/deployment/KotlinCoroutineIntegrationProcessor.java:197]: I can't find the FlowToPublisherHandler class anywhere.

Comment (geoand) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive-kotlin/deployment/src/main/java/io/quarkus/resteasy/reactive/kotlin/deployment/KotlinCoroutineIntegrationProcessor.java:197]: https://github.com/quarkusio/quarkus/blob/main/extensions/resteasy-reactive/quarkus-resteasy-reactive-kotlin/runtime/src/main/kotlin/org/jboss/resteasy/reactive/server/runtime/kotlin/FlowToPublisherHandler.kt

Comment (evanchooly) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive-kotlin/deployment/src/main/java/io/quarkus/resteasy/reactive/kotlin/deployment/KotlinCoroutineIntegrationProcessor.java:197]: Ah.  IDEA cache strikes again.  Thanks.  :)

Comment (geoand) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive-kotlin/deployment/src/main/java/io/quarkus/resteasy/reactive/kotlin/deployment/KotlinCoroutineIntegrationProcessor.java:197]: Yeah, the project is too big for IntelliJ to handle 😎

**Review by evanchooly — APPROVED** [review]

**Pull request merged**